bin/start_backend.sh

```diff
@@ -191,6 +191,11 @@ if [ ${RUN_CN} -eq 1 ]; then
     LOG_FILE=${LOG_DIR}/cn.out
 fi
 
+# enable DD profile
+if [ ${DD_PROFILING_ENABLED} == "true" ] && [ -f "${STARROCKS_HOME}/datadog/ddprof" ]; then
```

Comment:
1. `ENABLE_DATADOG_PROFILE` maybe a better name.
2. How much the overhead if running with this profile?
3. If the overhead is high, could we start to profile when we need?

Reply:
1. `DD_PROFILING_ENABLED` is the knob used by Datadog, prefer to use it directly instead of introducing another one;
2/3. that's a good question, from the [doc](https://docs.datadoghq.com/profiler/profiler_troubleshooting/java/#reduce-overhead-from-default-setup), it might be high with default settings. That's why we use `DD_PROFILING_ENABLED` as a switch, need more experience to balance the cost and benefit of continous profiling.

Reply:
> could we start to profile when we need?

AFAIK there's not an option to trigger a profiling on-demand. It's only on/off through life-cycle.

Reply:
@chaoyli I update the env to `ENABLE_DATADOG_PROFILE` as discussed to keep consitent. PTAL